hyperdome_client/threads.py

```
@@ -72,20 +72,232 @@ def run(self):
+
+    def __init__(self,
+                 server: Server,
+                 session: requests.Session,
+                 uid: str):
+        super(StartChatTask, self).__init__()
+        self.server = server
+        self.session = session
+        self.uid = uid
+
+    def run(self):
+        try:
+            self.signals.success.emit(start_chat(
+                self.server, self.session, self.uid))
+        except requests.RequestException:
+            self.signals.error.emit("Couldn't start a chat session")
+
+
+class GetMessagesTask(QtCore.QRunnable):
+    """
+    retrieve new messages on a fixed interval
+    """
+    signals = TaskSignals()
+
+    def __init__(self, session: requests.Session, server: Server, uid: str):
+        super(GetMessagesTask, self).__init__()
+        self.session = session
+        self.server = server
+        self.uid = uid
+
+    def run(self):
+        try:
+            new_messages = get_messages(self.server, self.session, self.uid)
+            self.signals.success.emit(new_messages)
+        except requests.HTTPError:
+            self.signals.error.emit("Counselor not in chat")
+        except requests.RequestException:
+            self.signals.error.emit("Error in get messages request")
+        except MethodNotAllowed:
+            self.signals.error.emit("not allowed")
+
+
+class ProbeServerTask(QtCore.QRunnable):
+    """
+    probe server for confirmation of hyperdome api
+    and api version compatibility
+    """
+    signals = TaskSignals()
+
+    def __init__(self, session: requests.Session, server: Server):
+        super(ProbeServerTask, self).__init__()
+        self.session = session
+        self.server = server
+
+    def run(self):
+        try:
+            status = probe_server(self.server, self.session)
+            if status:
+                raise Exception()
+
+            self.signals.success.emit('good')
+        except:
+            self.signals.error.emit('server incompatible')
```

Exception swallowing

---

I get it MOM